testplan/testing/multitest/driver/connection/base.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union
+from collections import defaultdict
+
+from testplan.common.utils.context import ContextValue
+
+
+class Direction(Enum):
+    connecting = "connecting"
+    listening = "listening"
+
+
+@dataclass
+class BaseConnectionInfo:
+    name: str  # name of the connection
+    service: str  # e.g. HTTP, TCP, FIX
+    protocol: str  # tcp, udp, file
+    identifier: Union[int, str, ContextValue]
+    direction: Direction
+
+    def to_dict(self):
+        return {
+            "service": self.service,
+            "protocol": self.protocol,
+            "identifier": self.identifier,
+            "direction": self.direction,
+        }
+
+    @property
+    def connection(self):
+        raise NotImplementedError
+
+
+class BaseDriverConnection:
+    """
+    Base class to show connection between drivers.
+    Each specific type (protocol) of connection should have its own subclass.
+    """
+
+    def __init__(self, driver_connection_info: BaseConnectionInfo):
```

Comment:
this constructor wasn't quite self-explaining at first sight, suggest having a attribute-initialising-only one and having a new class method for concrete logic needed